### PR TITLE
Add TLS config for local connections

### LINKED ISSUE #6: Add TLS termination
### Description

All communication between our services needs to be TLS terminated, including the cluster internal communication. As a result we can not rely on external PKI providers and thus must be able to supply a self signed CA root certificate unique for each deployment site.

### Proposed solution

Add options for each service to set a TLS certificate and corresponding CA root certificate.
These certificated should be mounted in at runtime.

### Definition of Done

RabbitMQ includes option for TLS termination.

### LINKED ISSUE #10: Enforce client verification
### Description
TLS is used to encrypt the traffic.
We currently use a self-signed certificate and do not enforce the verification of the clients.
This is a security advantage to ensure that only a given set of clients _can_ connect to the local and central brokers.

### Definition of Done
Besides the regular TLS encryption, the local message broker enforces the verification Python clients. Moreover, Central EGA enforces the verification of the local brokers.

### How to test
Connections should be accepted if and only if the containers are properly set up (ie using properly signed certificates).

### PR DESCRIPTION
**Pull request by jbygdell**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [X] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

Update rabbitmq config to enable TLS for local connections.
The following restrictions apply; 
* TLS version 1.2 or above is required.

These certificates are not used when connecting to CEGA rabbitmq, those changes are comming in a separate PR.

### Changes made:

1) Add TLS settings to rabbitmq.conf
2) replace `sed` in the entrypoint script since the cega connection URI will contain ampersand characters.

### Related issues:

Fixes #6
Fixes #10

### CODE REVIEW

File: rabbitmq.conf [hunk 1]
```diff
@@ -1,4 +1,11 @@
 listeners.tcp.default = 5672
+listeners.ssl.default = 5671
+ssl_options.cacertfile = /etc/rabbitmq/ssl/ca-root.pem
+ssl_options.certfile = /etc/rabbitmq/ssl/mq-server.pem
+ssl_options.keyfile = /etc/rabbitmq/ssl/mq-server-key.pem
+ssl_options.verify = verify_peer
```
Comment (silverdaz) [inline rabbitmq.conf]: This setting is useful to make the server verify the client certificate.
However, this is only interesting for the CentralEGA server at the moment.
We don't need to enable it for the local broker yet.
The python/java clients do not have any certificate yet, so they won't manage to connect.

Leave the (fake) CentralEGA settings out of this repo, we can set them somewhere else.

Comment (silverdaz) [reply rabbitmq.conf]: eh.... not sure why github shows the whole block. I meant to comment on line 6 only.
```ssl_options.verify = verify_peer```

File: entrypoint.sh [hunk 2]
```diff
@@ -9,4 +9,17 @@ sed -i 's%MQ_PASSWORD_HASH%'${MQ_PASSWORD_HASH}'%g' /etc/rabbitmq/definitions.js
...
+# Generating the SSL certificate + key
+openssl req -x509 -newkey rsa:2048 \
+    -keyout /etc/rabbitmq/ssl/mq-server.key -nodes \
+    -out /etc/rabbitmq/ssl/mq-server.cert -sha256 \
+    -days 1000 -subj ${SSL_SUBJ}
+else
+# Otherwise use the injected ones.
+echo "Using the injected certificate/privatekey pair" 
+fi
+chmod 600 /etc/rabbitmq/ssl/mq-server.key
```
Comment (silverdaz) [inline entrypoint.sh]: Not sure the rabbitmq-server runs under root.
Isn't it running under rabbitmq user?

Comment (jbygdell) [reply entrypoint.sh]: nope, root.
But there are a few issues still.

**Comment by jbygdell** [comment]
Closing for now until the kinks have been ironed out.

**Comment by jbygdell** [comment]
Closing for now until the kinks have been ironed out.

**Pull request merged**